pkg/controller/volume/attach_detach_controller.go

```diff
@@ -90,7 +136,95 @@ func NewAttachDetachController(
 		DeleteFunc: adc.nodeDelete,
 	})
 
-	return adc
+	if err := adc.volumePluginMgr.InitPlugins(plugins, adc); err != nil {
+		return nil, fmt.Errorf("Could not initialize volume plugins for Attach/Detach Controller: %+v", err)
+	}
+
+	adc.desiredStateOfWorld = cache.NewDesiredStateOfWorld(&adc.volumePluginMgr)
+	adc.actualStateOfWorld = cache.NewActualStateOfWorld(&adc.volumePluginMgr)
```

Comment:
That was my original plan, but I when I started implementing it, turns out it's much easier to maintain two independent data structures especially because we have pods "trickling" in and not all available at once.